.github/workflows/cloud-tpu-ci-nightly.yml

```diff
@@ -0,0 +1,59 @@
+name: Cloud TPU nightly
+
+on:
+  schedule:
+    - cron: "0 14 * * *"  # daily at 7am PST
+  workflow_dispatch:  # allows triggering the workflow run manually
+
+# Should also be set to read-only in the project settings.
+permissions:
+  contents: read
+
+jobs:
+  cloud-tpu-test:
+    runs-on: v4-8
```

Comment:
I think it's "strongly recommended" to include "self-hosted" here:

https://docs.github.com/en/actions/hosting-your-own-runners/using-self-hosted-runners-in-a-workflow#using-self-hosted-runners-in-a-workflow